# IS 362 - Project 1: Airline Arrival Delays

## Introduction

This project analyzes arrival delays for two airlines across five destinations.

The data includes the number of flights that arrived on time and the number of flights that were delayed for each airline and destination. The data is stored in a CSV file and will be read into pandas for analysis.

The goal is to compare the arrival delays for the two airlines and describe the findings using pandas.

In [2]:
import pandas as pd

data = pd.read_csv("Nathan Fernandez - IS 362 - Project 1.csv")

print (data)

  Airline    Destination  On Time  Delayed
0  ALASKA    Los Angeles      497       62
1  ALASKA        Phoenix      221       12
2  ALASKA      San Diego      212       20
3  ALASKA  San Francisco      503      102
4  ALASKA        Seattle     1841      305
5  AMWEST    Los Angeles      694      117
6  AMWEST        Phoenix     4840      415
7  AMWEST      San Diego      383       65
8  AMWEST  San Francisco      320      129
9  AMWEST        Seattle      201       61


## Reading the CSV Data

The CSV file contains information about two airlines, ALASKA and AMWEST, across five destinations.

The data includes the number of flights that arrived on time and the number of flights that were delayed. The CSV file was read into pandas as a DataFrame so that the information can be examined and analyzed.

In [3]:
airline_totals = data.groupby("Airline")[["On Time", "Delayed"]].sum()

print (airline_totals)

         On Time  Delayed
Airline                  
ALASKA      3274      501
AMWEST      6438      787


In [4]:
delay_percentage = (
    airline_totals["Delayed"] /
    (airline_totals["On Time"] + airline_totals["Delayed"])
) * 100

print (delay_percentage)

Airline
ALASKA    13.271523
AMWEST    10.892734
dtype: float64


## Comparing Overall Delay Rates

The total number of delayed flights is higher for AMWEST, but AMWEST also has more total flights.

To make a fairer comparison, the percentage of delayed flights was calculated for each airline by dividing the number of delayed flights by the total number of flights.

Based on the calculated percentages, ALASKA had a higher percentage of delayed flights than AMWEST in this dataset.

In [5]:
destination_totals = data.groupby("Destination")[["On Time", "Delayed"]].sum()

destination_delay_rates = (
    destination_totals["Delayed"] /
    (destination_totals["On Time"] + destination_totals["Delayed"])
) * 100

print (destination_delay_rates)

Destination
Los Angeles      13.065693
Phoenix           7.780612
San Diego        12.500000
San Francisco    21.916509
Seattle          15.199336
dtype: float64


## Comparing Delay Rates by Destination

The combined delay rate varies across the five destinations.

Los Angeles had a delay rate of around 13.07%, Phoenix had 7.78%, San Diego had 12.50%, San Francisco had 21.92%, and Seattle had 15.20%.

These results show that the percentage of delayed flights differs depending on the destination. The destination-level analysis provides additional context for comparing the two airlines.

In [6]:
data["Delay Rate"] = (
    data["Delayed"] /
    (data["On Time"] + data["Delayed"])
) * 100

print (data[["Airline", "Destination", "Delay Rate"]])

  Airline    Destination  Delay Rate
0  ALASKA    Los Angeles   11.091234
1  ALASKA        Phoenix    5.150215
2  ALASKA      San Diego    8.620690
3  ALASKA  San Francisco   16.859504
4  ALASKA        Seattle   14.212488
5  AMWEST    Los Angeles   14.426634
6  AMWEST        Phoenix    7.897241
7  AMWEST      San Diego   14.508929
8  AMWEST  San Francisco   28.730512
9  AMWEST        Seattle   23.282443


In [8]:
comparison = data.pivot(
    index="Destination",
    columns="Airline",
    values="Delay Rate"
).round(2)

print (comparison)

Airline        ALASKA  AMWEST
Destination                  
Los Angeles     11.09   14.43
Phoenix          5.15    7.90
San Diego        8.62   14.51
San Francisco   16.86   28.73
Seattle         14.21   23.28


## Comparing Airlines by Destination

The delay rates can be compared directly for each destination. ALASKA had a lower delay rate than AMWEST at all five destinations in this dataset.

For ALASKA, the delay rates ranged from 5.15% in Phoenix to 16.86% in San Francisco. For AMWEST, the delay rates ranged from 7.90% in Phoenix to 28.73% in San Francisco.

The results show that delay rates varied by both airline and destination. San Francisco had the highest delay rate for both airlines, while Phoenix had the lowest delay rate for both airlines.

## Conclusion

This analysis compared arrival delay rates for ALASKA and AMWEST across five destinations.

Overall, ALASKA had a delay rate of roughly 13.27%, while AMWEST had a delay rate of roughly 10.89% when all destinations were combined. However, when the airlines were compared at each individual destination, ALASKA had a lower delay rate than AMWEST at all five destinations.

The destination-level results also showed differences in delay rates. San Francisco had the highest delay rate for both airlines, while Phoenix had the lowest delay rate for both airlines.

These results demonstrate why comparing data at both the overall and destination levels can provide a more complete understanding of the differences in the dataset.